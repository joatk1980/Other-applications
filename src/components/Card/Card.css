.Weather-card {
  border: 1px solid black;
  width: 100%;
  border-radius: 6px;
  padding: 16px;
  font-size: 14px;
  margin-bottom: 20px;
  margin-top: 20px;
}

.day {
  font-size: 16px;
  font-weight: bold;
  color: rgb(17, 17, 17);
}

.description {
  font-size: 12px;
  color: black;
}

.temp {
  font-size: 12px;
  color: black;
}

.windspeed {
  font-size: 12px;
  color: black;
}

.icon {
  max-width: 100px;
}

@media only screen and (min-width: 500px) {
  .Weather-card {
    width: 15%;
    min-width: 200px;
  }
}
