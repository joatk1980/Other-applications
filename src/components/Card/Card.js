import React from "react";
import "./Card.css";
import getWeatherCondition from "../helpers/getWeatherConditions";
import transformDate from "../helpers/transformDate/index";

const Card = ({ day }) => {
  console.log("day", day);
  const conditionDesc = getWeatherCondition(day.condition)[0];
  const icon = getWeatherCondition(day.condition)[1];
  const date = transformDate(day.date);

  return (
    <div className="Weather-card">
      <h3 className="day">
        {day.dayOfWeek} / {date}
      </h3>
      <img className="icon" src={icon} alt="Weather Icon" />
      <p className="condition">{conditionDesc}</p>
      <p>
        Temp: <strong>{day.temperatureMax} Â°C</strong>
      </p>
      <p>
        Wind: <strong>{day.windSpeed} km/h</strong>
      </p>
    </div>
  );
};

export default Card;
