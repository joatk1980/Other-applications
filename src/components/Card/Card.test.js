import renderer from "react-test-renderer";
import Card from "./Card";

it("Renders the Card", () => {
  const day = {
    date: "2024-06-26",
    dayOfWeek: "Wednesday",
    condition: 3,
    temperatureMax: 29,
    temperatureMin: 16,
    windSpeed: 12,
  };
  const component = renderer.create(<Card day={day} />);
  expect(component).toMatchSnapshot();
});
