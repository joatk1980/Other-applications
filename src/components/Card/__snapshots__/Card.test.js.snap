// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the Card 1`] = `
<div
  className="Weather-card"
>
  <h3
    className="day"
  >
    Wednesday
     / 
    26th June
  </h3>
  <img
    alt="Weather Icon"
    className="icon"
    src="Sun_cloud.png"
  />
  <p
    className="condition"
  >
    Mainly clear, partly cloudy, and overcast
  </p>
  <p>
    Temp: 
    <strong>
      29
       Â°C
    </strong>
  </p>
  <p>
    Wind: 
    <strong>
      12
       km/h
    </strong>
  </p>
</div>
`;
